xrm/xrm.d.ts

```diff
@@ -437,49 +445,7 @@ declare namespace Xrm
          */
         refreshRibbon(): void;
 
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"ERROR"}   level       An error message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "ERROR", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"WARNING"} level       A warning message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "WARNING", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {"INFO"}    level       An informational message.
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, false if it fails.
-         */
-        setFormNotification( message: string, level: "INFO", uniqueId: string ): boolean;
-
-        /**
-         * Sets a form-level notification.
-         *
-         * @param   {string}    message     The message.
-         * @param   {string}    level       The level, as either "ERROR", "WARNING", or "INFO".
-         * @param   {string}    uniqueId    A unique identifier for the message.
-         *
-         * @return  true if it succeeds, otherwise false.
-         */
-        setFormNotification( message: string, level: string, uniqueId: string ): boolean;
+        setFormNotification(message: string, level: Page.ui.FormNotificationLevel, uniqueId: string ): boolean;
```

Comment:
this will break these scenarions:

```ts
var level: "ERROR";

ui.setFormNotification("message", level, "id");  /// string is not assignable toFormNotificationLevel
```